pkg/gui/diff_context_size.go

```diff
@@ -0,0 +1,50 @@
+package gui
+
+func getRefreshFunction(gui *Gui) func()error {
+	key := gui.currentStaticContext().GetKey()
+
+	if key == FILES_CONTEXT_KEY {
+		return gui.refreshFilesAndSubmodules
+	} else if key == MAIN_STAGING_CONTEXT_KEY {
+		return func() error {
+			selectedLine := gui.Views.Secondary.SelectedLineIdx()
+			return gui.handleRefreshStagingPanel(false, selectedLine)
+		}
+	} else if key == COMMIT_FILES_CONTEXT_KEY || key == MAIN_PATCH_BUILDING_CONTEXT_KEY {
+		return func() error {
+			selectedLine := gui.Views.Secondary.SelectedLineIdx()
+			return gui.handleRefreshPatchBuildingPanel(selectedLine)
+		}
+	} else if key == STASH_CONTEXT_KEY {
+		return gui.handleStashEntrySelect
+	} else if key == BRANCH_COMMITS_CONTEXT_KEY {
+		return gui.handleCommitSelect
+	} else if key == SUB_COMMITS_CONTEXT_KEY {
+		return gui.handleSubCommitSelect
+	}
+
+	return nil
+}
+
+func (gui *Gui) IncreaseContextInDiffView() error {
+	refresh := getRefreshFunction(gui)
+
+	if refresh != nil {
+		gui.Config.GetUserConfig().Git.DiffContextSize = gui.Config.GetUserConfig().Git.DiffContextSize + 1
+		return refresh()
+	}
+
+	return nil
+}
+
+func (gui *Gui) DecreaseContextInDiffView() error {
+	refresh := getRefreshFunction(gui)
+	old_size := gui.Config.GetUserConfig().Git.DiffContextSize
+
+	if refresh != nil && old_size > 1 {
+		gui.Config.GetUserConfig().Git.DiffContextSize = old_size - 1
```

Comment:
we don't change the user's config object directly elsewhere, and I'm not entirely sure whether we should try to avoid it. Changing it directly spares us from having to maintain this state elsewhere, but I wonder if there's value in keeping the user's config separate from whatever's being changed at runtime. I'm just thinking out loud here, I'm currently leaning towards this implementation being okay

Reply:
That's good to know, I didn't notice. My initial thought was that it might be nice for a user to set a default value in their config, but in practice I found it makes more sense to just use the shortcuts.
Would you have a suggestion as to where the `DiffContextSize` member could live?